src/Raven.Client/Documents/Session/IDocumentSession.IncludeCounters.cs

```diff
@@ -0,0 +1,36 @@
+﻿//-----------------------------------------------------------------------
+// <copyright file="IDocumentSession.cs" company="Hibernating Rhinos LTD">
+//     Copyright (c) Hibernating Rhinos LTD. All rights reserved.
+// </copyright>
+//-----------------------------------------------------------------------
+
+using Raven.Client.Documents.Session.Loaders;
+
+namespace Raven.Client.Documents.Session
+{
+    /// <summary>
+    ///     Interface for document session
+    /// </summary>
+    public partial interface IDocumentSession
```

Comment:
This I don't like.
We already have a top level "taint" with `CountersFor`, I don't want to add `IncludeCounters` there at this level.
@ppekrol Any suggestions about the kind of API we should strive for?

Reply:
Let me think about it...

Reply:
I do not think that we have other choice here. We are in the context of the document here and if we want to download the counters in a single call then there is no other option since `Load` is materializing the entity right away.

Reply:
How about `Includes` property? 
Then we hang everything on top of that?

Reply:
Or:
```
session.Includes<Order>("orders/1-A")
    .Counter("likes")
    .Document(x=>x.Employee)
    .Load();
```

Reply:
Maybe something like that?

```csharp
session.Load<Order>(
    "orders/1-A", 
    includes => includes
        .Counter("likes")
        .Documents(x => x.Employee));
```

Reply:
Ohhhh. I _like_ this, yes.

Reply:
Would we do the same in `Query`?

Reply:
```csharp
session.Query<Company>()
    .Include(includes => includes
        .Counter("likes")
        .Documents(x => x.Employee))
    .Where(x => x.Name == "John")
    .ToList();
```

Reply:
Looks good to me, but might be hard to retrofit with what we already have, no?

Reply:
Should be doable, but we cannot remove old 'Include' methods until 5.0.

Reply:
```csharp
        public static IRavenQueryable<TResult> Include<TResult>(this IQueryable<TResult> source, Action<IFacetBuilder<TResult>> builder)
        {
            return null;
        }
```

```csharp
                    var orders = session.Query<Order>()
                        .Customize(x => x.WaitForNonStaleResults())
                        .Include(x => x.AllResults())
                        .Include(x => x.CustomerId)
                        .ToList();
```

compiles and works. Although ReSharper is going crazy here with Intellisense.

Reply:
I'm in awe. Poor C#.

Reply:
Although, we can't use fluent type here, or can we use overload based on the `Func<IncludeBuilder>` ?

Reply:
I think it could be `Action<IncludeBuilder<TResult>>`.

Reply:
changed to

Reply:
create a ticket for 
```
session.Query<Company>()
    .Include(includes => includes
        .Counter("likes")
        .Documents(x => x.Employee))
    .Where(x => x.Name == "John")
    .ToList();
``` ?

Reply:
When you will address those few cosmetic comments that I just wrote, then we will be able to merge it and you can continue work on the query under the same issue number.